src/layers/core/point-density-hexagon-layer/point-density-hexagon-layer.js

```diff
@@ -110,15 +150,26 @@ export default class PointDensityHexagonLayer extends Layer {
 
   _onGetSublayerColor(cell) {
     const {colorRange} = this.props;
-    const colorDomain = this.props.colorDomain || this.state.countRange;
+    const {countDomainByPercentile} = this.state;
+    const colorDomain = this.props.colorDomain || countDomainByPercentile;
+    const count = cell.points.length;
+
+    const color = ordinalScale(colorDomain, colorRange, count);
 
-    return ordinalScale(colorDomain, colorRange, cell.points.length);
+    // if cell count is outside domain, set alpha to 0
+    const alpha = count >= countDomainByPercentile[0] && count <= countDomainByPercentile[1] ?
+      (Number.isFinite(color[3]) ? color[3] : 255) : 0;
+
+    return color.concat([alpha]);
   }
 
   _onGetSublayerElevation(cell) {
-    const {elevationRange} = this.props;
-    const elevationDomain = this.props.elevationDomain || [0, this.state.countRange[1]];
-    return linearScale(elevationDomain, elevationRange, cell.points.length);
+    const {elevationDomain, elevationRange} = this.props;
+    const {sortedCounts} = this.state;
+
+    // elevation is not affected by percentile
```

Comment:
Maybe, I find have both color and elevation affected by percentile is very confusing. I would rather wait until user request to add it